### PR TITLE
Announcement backward compatibility

### PR DESCRIPTION
**Pull request by saraycp**
Adapt announcement endpoints to act as status messages. Announcement functionality has been removed but not its endpoints to keep backward compatibility. Instead, they will end up in status messages actions and views.
    
Modifications:
  - Adapt status message policy.
  - Rename 'message' to 'status_message'.
  - Add update action to status messages to be able to keep update
    announcement's endpoint.
  - Use absolute routes inside status messages partials.

This PR shouldn't be merged until #9624 is merged and the migration is run.
This is part of a series of PRs including: PR #9590, #9613, #9624.

Co-authored-by: Victor Pereira <vpereira@suse.com>

### CODE REVIEW

File: src/api/app/controllers/announcements_controller.rb [hunk 1]
```diff
@@ -1,73 +1,64 @@
 class AnnouncementsController < ApplicationController
-  before_action :set_announcement, only: [:show, :update, :destroy]
+  before_action :set_status_message, only: [:show, :update, :destroy]
```
Comment (hennevogel) [inline src/api/app/controllers/announcements_controller.rb:2]: Are you missing `require_admin`?

File: ReleaseNotes-2.11 [hunk 2]
```diff
@@ -69,3 +69,4 @@ Breaking API changes:
 =====================
 
  * Status messages endpoints. Check the API documentation for the new format.
+ * Announcement outputs have changed. Check the API documentation for the new format.
```
Comment (hennevogel) [inline ReleaseNotes-2.11:72]: No need to mention the changed output. Mention that  `/announcement` is a deprecated API and will be removed in the next version.

File: docs/api/api/api.txt [hunk 3]
```diff
@@ -28,38 +28,41 @@ XmlResult: about
 
 == Announcements
 
+WARNING: Announcements will be deprecated soon. Please use Status Message endpoints
+ with 'announcement' severity instead.
```
Comment (hennevogel) [inline docs/api/api/api.txt:32]: This belongs into the Changelog, all documentation here can be removed.

File: docs/api/api/status_messages.xsd [hunk 4]
```diff
@@ -1,44 +1,38 @@
...
-      The value for the attribute 'severity' (0-3) has the following meaning:
-        0: informal message
-        1: status changed to good with this message (green traffic lights displayed)
-        2: status changed to mid critical (yellow traffic lights displayed)
-        3: status changed to critical (red traffic lights displayed)
+      The value for the attribute 'severity' has the following meaning:
+        information: informal message
+        green: status changed to good with this message (green traffic lights displayed)
+        yellow: status changed to mid critical (yellow traffic lights displayed)
+        red: status changed to critical (red traffic lights displayed)
```
Comment (hennevogel) [inline docs/api/api/status_messages.xsd:15]: There are no traffic lights displayed anywhere anymore.

Comment (DavidKang) [reply docs/api/api/status_messages.xsd:15]: true!

File: src/api/app/controllers/announcements_controller.rb (hunk 1 above)
Comment (DavidKang) [reply src/api/app/controllers/announcements_controller.rb:2]: IMO, this is covered Pundit (status_message_policies). unless I miss something.

**Comment by saraycp** [comment]
@hennevogel feedback addressed

File: docs/api/api/api.txt [hunk 5]
```diff
@@ -32,34 +32,34 @@ GET /announcements
 
   Get a list of all announcements.
```
Comment (hennevogel) [inline docs/api/api/api.txt:33]: You should remove the whole section

File: ReleaseNotes-2.11 [hunk 6]
```diff
@@ -69,3 +69,4 @@ Breaking API changes:
 =====================
 
  * Status messages endpoints. Check the API documentation for the new format.
+ * The API endpoint '/announcements' is deprecated and will be removed in the next version.
```
Comment (hennevogel) [inline ReleaseNotes-2.11:72]: Please mention what replaces it.

File: docs/api/api/api.txt (hunk 5 above)
Comment (saraycp) [reply docs/api/api/api.txt:33]: Oh! I understand now. So no documentation at all about announcements. :+1:

**Pull request merged**